File: SchoolManagementApi/DTOs/Auth/LoginResponseDto.cs
namespace SchoolManagementApi.DTOs.Auth;

public enum UserRole { Teacher, Student }

public class LoginResponseDto
{
    public int Id { get; set; }
    public string Email { get; set; } = null!;
    public string Name { get; set; } = null!;
    public string Surname { get; set; } = null!;
    public UserRole Role { get; set; } 
}

File: SchoolManagementApiNode/src/auth/dto/login-response.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { UserRole } from './login.dto';

export class LoginResponseDto {
  @ApiProperty()
  id: number;

  @ApiProperty()
  email: string;

  @ApiProperty()
  name: string;

  @ApiProperty()
  surname: string;

  @ApiProperty({ enum: UserRole })
  role: UserRole;
}
